src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -12,67 +12,48 @@
 'use strict';
 
 describe('DOMPropertyOperations', () => {
-  var DOMPropertyOperations;
-  var DOMProperty;
-  var ReactDOMComponentTree;
+  var React;
+  var ReactDOM;
 
   beforeEach(() => {
     jest.resetModules();
-    require('ReactDOMInjection');
-
-    // TODO: can we express this test with only public API?
-    DOMPropertyOperations = require('DOMPropertyOperations');
-    DOMProperty = require('DOMProperty');
-    ReactDOMComponentTree = require('ReactDOMComponentTree');
+    React = require('react');
+    ReactDOM = require('react-dom');
   });
 
   describe('setValueForProperty', () => {
-    var stubNode;
-    var stubInstance;
-
-    beforeEach(() => {
-      stubNode = document.createElement('div');
-      stubInstance = {_debugID: 1};
-      ReactDOMComponentTree.precacheNode(stubInstance, stubNode);
-    });
-
     it('should set values as properties by default', () => {
-      DOMPropertyOperations.setValueForProperty(stubNode, 'title', 'Tip!');
-      expect(stubNode.title).toBe('Tip!');
+      var container = document.createElement('div');
+      ReactDOM.render(<div title="Tip!" />, container);
+      expect(container.firstChild.title).toBe('Tip!');
     });
 
     it('should set values as attributes if necessary', () => {
-      DOMPropertyOperations.setValueForProperty(stubNode, 'role', '#');
-      expect(stubNode.getAttribute('role')).toBe('#');
-      expect(stubNode.role).toBeUndefined();
+      var container = document.createElement('div');
+      ReactDOM.render(<div role="#" />, container);
+      expect(container.firstChild.getAttribute('role')).toBe('#');
+      expect(container.firstChild.role).toBeUndefined();
     });
```

Comment:
Just thinking: I didn't see a test that covers attributes that require property assignment. I think it would be covered by a test that uses `multiple` with a select, but the behavior is probably covered by ReactDOMSelect-test.